### DISCUSSION TITLE
I want to get an battery percentage of remote machine

### DISCUSSION THREAD

**Discussion by scitara-shashank** [justcoding121/Titanium-Web-Proxy#959] [Q&A] [answered]
Hi Everyone,

I have an requirement where I want to get an information from remote client machine about the battery percentage and it's machine name.

Can you please suggest me someway to do it.

I understand we can achieve by sending the headers too, but I want to avoid to reading the encrypted connections

**Comment by justcoding121** [comment, marked as answer]
This isn't something a proxy can do on its own. Titanium only sees HTTP(S) traffic ? it has no access to battery %, machine name, or other OS info on the remote client.

A few realistic options:
1. Run a small agent/service on the client that reports that info (over HTTP to your app, or as a custom header if you control the client).
2. Use OS remote management (WMI/WinRM, MDM, etc.) ? outside this library.
3. If the client is your own app, just query `System.Windows.Forms.SystemInformation` / `Windows.Devices.Power` / similar locally and send it yourself.

Decrypting TLS won't help either ? that only gives you HTTP content, not hardware state.

Closing since this falls outside what the proxy can provide. Feel free to reopen if you have a proxy-specific angle.
